src/csharp/Extensions/Microsoft.Spark.Extensions.FileSystem/JvmReferenceFileSystem.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+
+namespace Microsoft.Spark.Extensions.FileSystem
+{
+    /// <summary>
+    /// <see cref="FileSystem"/> implementation that wraps a corresponding FileSystem object in the JVM.
+    /// </summary>
+    public class JvmReferenceFileSystem : FileSystem, IJvmObjectReferenceProvider
```

Comment:
Why is `JvmReference` public?

Reply:
I am thinking that we should just put APIs into `FileSystem` since `get` handles the getting the right concrete class. In what scenarios, do you see we need a concrete implementation of `FileSystem` other than hadoop filesystem?

Reply:
And just name the package as `Microsoft.Spark.Extensions.Hadoop.FileSystem` so that there is no confusion; i.e., we are just wrapping `org.apache.hadoop.fs.FileSystem` more or less.

Reply:
Sure, I'll just make `FileSystem` concrete.  I was just trying to keep it as similar to the JVM implementation as possible.  But I suppose we won't have any POCO `FileSystem` implementations.

Reply:
Made `FileSystem` concrete and removed `JvmReferenceFileSystem`.

Reply:
Changed implementation to use `Configuration`.